### ISSUE TITLE
Python DirectML silently fails

### ISSUE THREAD

**Issue by CashWilliams** [microsoft/onnxruntime-genai#1092]
**Describe the bug**
When attempting to follow https://onnxruntime.ai/docs/genai/tutorials/phi3-python.html, no output is ever produced.

**To Reproduce**

Following https://onnxruntime.ai/docs/genai/tutorials/phi3-python.html

```
(venv) PS C:\Users\CashW\src\onnx> curl https://raw.githubusercontent.com/microsoft/onnxruntime-genai/main/examples/python/phi3-qa.py -o phi3-qa.py
(venv) PS C:\Users\CashW\src\onnx> python phi3-qa.py -m directml\directml-int4-awq-block-128
Input: tell me a joke
(venv) PS C:\Users\CashW\src\onnx> python phi3-qa.py -m directml\directml-int4-awq-block-128 --verbose
Loading model...
Model loaded
Tokenizer created

Input: tell me a joke
(venv) PS C:\Users\CashW\src\onnx>
```

**Expected behavior**
Model output would be printed, or at least some error.

**Desktop (please complete the following information):**
 Windows 11

```
(venv) PS C:\Users\CashW\src\onnx> pip list
Package                    Version
-------------------------- ---------
certifi                    2024.8.30
charset-normalizer         3.4.0
colorama                   0.4.6
coloredlogs                15.0.1
filelock                   3.16.1
flatbuffers                24.3.25
fsspec                     2024.10.0
huggingface-hub            0.26.2
humanfriendly              10.0
idna                       3.10
inquirerpy                 0.3.4
mpmath                     1.3.0
numpy                      2.1.3
onnxruntime-directml       1.20.1
onnxruntime-genai-directml 0.5.1
packaging                  24.2
pfzy                       0.3.4
pip                        24.2
prompt_toolkit             3.0.48
protobuf                   5.29.0rc3
pyreadline3                3.5.4
PyYAML                     6.0.2
requests                   2.32.3
sympy                      1.13.3
tqdm                       4.67.0
typing_extensions          4.12.2
urllib3                    2.2.3
wcwidth                    0.2.13
```

```
(venv) PS C:\Users\CashW\src\onnx> nvidia-smi.exe
Fri Nov 22 12:46:37 2024
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 560.94                 Driver Version: 560.94         CUDA Version: 12.6     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                  Driver-Model | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA GeForce RTX 4080 ...  WDDM  |   00000000:01:00.0 Off |                  N/A |
| N/A   39C    P0             22W /  175W |       0MiB /  12282MiB |      0%      Default |
|                                         |                        |                  N/A |
+-----------------------------------------+------------------------+----------------------+

+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI        PID   Type   Process name                              GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
```

**Comment by CashWilliams** [comment]
I tried adding a try/catch as well as some additional print statements:

```
try:
    generator = og.Generator(model, params)
except:
    print("Generator creation failed")
print("Generator created")
```

Neither line prints, it just seems to silently kill the python process somewhere within the `og.Generator()` call.

**Comment by baijumeswani** [comment]
This is a known issue with the release 0.5.1. The issue was fixed in https://github.com/microsoft/onnxruntime-genai/pull/1070 but a patch release has yet to be published.

In the meantime, could you try downgrading to version 0.5.0? Apologies for the inconvenience.

**Comment by CashWilliams** [comment]
Thanks for the quick response!!

I looked in open issues, didn't think to look in recently closed 😄 

downgrading `onnxruntime-genai-directml` to `0.5.0` fixed the issue.


```
(venv) PS C:\Users\CashW\src\onnx> pip install --upgrade onnxruntime-genai-directml==0.5.0
Collecting onnxruntime-genai-directml==0.5.0
  Downloading onnxruntime_genai_directml-0.5.0-cp312-cp312-win_amd64.whl.metadata (682 bytes)
Requirement already satisfied: numpy>=1.21.6 in c:\users\cashw\src\onnx\venv\lib\site-packages (from onnxruntime-genai-directml==0.5.0) (2.1.3)
Requirement already satisfied: onnxruntime-directml>=1.20.0 in c:\users\cashw\src\onnx\venv\lib\site-packages (from onnxruntime-genai-directml==0.5.0) (1.20.1)
Requirement already satisfied: coloredlogs in c:\users\cashw\src\onnx\venv\lib\site-packages (from onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (15.0.1)
Requirement already satisfied: flatbuffers in c:\users\cashw\src\onnx\venv\lib\site-packages (from onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (24.3.25)
Requirement already satisfied: packaging in c:\users\cashw\src\onnx\venv\lib\site-packages (from onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (24.2)
Requirement already satisfied: protobuf in c:\users\cashw\src\onnx\venv\lib\site-packages (from onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (5.29.0rc3)
Requirement already satisfied: sympy in c:\users\cashw\src\onnx\venv\lib\site-packages (from onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (1.13.3)
Requirement already satisfied: humanfriendly>=9.1 in c:\users\cashw\src\onnx\venv\lib\site-packages (from coloredlogs->onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (10.0)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in c:\users\cashw\src\onnx\venv\lib\site-packages (from sympy->onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (1.3.0)
Requirement already satisfied: pyreadline3 in c:\users\cashw\src\onnx\venv\lib\site-packages (from humanfriendly>=9.1->coloredlogs->onnxruntime-directml>=1.20.0->onnxruntime-genai-directml==0.5.0) (3.5.4)
Downloading onnxruntime_genai_directml-0.5.0-cp312-cp312-win_amd64.whl (1.9 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.9/1.9 MB 8.2 MB/s eta 0:00:00
Installing collected packages: onnxruntime-genai-directml
  Attempting uninstall: onnxruntime-genai-directml
    Found existing installation: onnxruntime-genai-directml 0.5.1
    Uninstalling onnxruntime-genai-directml-0.5.1:
      Successfully uninstalled onnxruntime-genai-directml-0.5.1
Successfully installed onnxruntime-genai-directml-0.5.0

[notice] A new release of pip is available: 24.2 -> 24.3.1
[notice] To update, run: python.exe -m pip install --upgrade pip
(venv) PS C:\Users\CashW\src\onnx> python phi3-qa.py -m directml\directml-int4-awq-block-128 --verbose
Loading model...
Model loaded
Tokenizer created

Input: tell me a joke
Generator created
Running generation loop ...

Output:  Certainly! Here's a light-hearted joke for you:

Why don't scientists trust atoms? Because they make up everything!

Remember, humor can be subjective, so I hope you found this joke amusing. If you're interested in more, feel free to ask!
```
